src/main/java/com/yahoo/bullet/pubsub/rest/RESTPubSub.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright 2018, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.pubsub.rest;
+
+import com.yahoo.bullet.common.BulletConfig;
+import com.yahoo.bullet.pubsub.PubSub;
+import com.yahoo.bullet.pubsub.PubSubException;
+import com.yahoo.bullet.pubsub.Publisher;
+import com.yahoo.bullet.pubsub.Subscriber;
+import lombok.extern.slf4j.Slf4j;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.AsyncHttpClientConfig;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Slf4j
+public class RESTPubSub extends PubSub {
+    private static final int NO_TIMEOUT = -1;
+    public static final int OK_200 = 200;
+    public static final int NO_CONTENT_204 = 204;
+
+    /**
+     * Create a RESTPubSub from a {@link BulletConfig}.
+     *
+     * @param config The config.
+     * @throws PubSubException
+     */
+    public RESTPubSub(BulletConfig config) throws PubSubException {
+        super(config);
+        this.config = new RESTPubSubConfig(config);
+    }
+
+    @Override
+    public Publisher getPublisher() {
+        if (context == Context.QUERY_PROCESSING) {
+            return new RESTResultPublisher(config, getClient());
+        } else {
+            return new RESTQueryPublisher(config, getClient());
+        }
+    }
+
+    @Override
+    public List<Publisher> getPublishers(int n) {
+        return IntStream.range(0, n).mapToObj(i -> getPublisher()).collect(Collectors.toList());
+    }
+
+    @Override
+    public Subscriber getSubscriber() {
+        int maxUncommittedMessages = config.getAs(RESTPubSubConfig.MAX_UNCOMMITTED_MESSAGES, Number.class).intValue();
+        RESTPubSubConfig pubsubConfig = new RESTPubSubConfig(config);
```

Comment:
Why do you create a new config? Can you just use config instead since it's already RESTPubSubConfig type?

Reply:
config is a BulletConfig.  But I should be getting maxUncommittedMessages from the RESTPubSubConfig.  I'll fix that.